bin/git-bulk

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env bash
+invers=`tput rev`
+reset=`tput sgr0`
+txtbld=$(tput bold)             
+bldred=${txtbld}$(tput setaf 1) 
+
+# default option settings
+guardedmode=false
+singlemode=false
+allwsmode=false
+
+#
+# print usage message
+#
+usage() {
+  echo 1>&2 "usage: git bulk [-g] ([-a]|[-w <ws-name>]) <git command>"
+  echo 1>&2 "       git bulk --addworkspace <ws-name> <ws-root-directory>"
+  echo 1>&2 "       git bulk --removeworkspace <ws-name>"
+  echo 1>&2 "       git bulk --addcurrent <ws-name>"
+  echo 1>&2 "       git bulk --purge"
+  echo 1>&2 "       git bulk --listall"
+}
+
+# add another workspace to global git config
+function addworkspace { git config --global bulkworkspaces.$wsname "$wsdir"; }
+
+# add current directory
+function addcurrent { git config --global bulkworkspaces.$wsname "$PWD"; }
+
+# remove workspace from global git config
+function removeworkspace { checkWSName && git config --global --unset bulkworkspaces.$wsname; }
+
+# remove workspace from global git config
+function purge { git config --global --remove-section bulkworkspaces; }
+
+# list all current workspace locations defined
+function listall { git config --global --get-regexp bulkworkspaces; }
+
+# guarded execution of a git command in one specific repository
+function guardedExecution () {
+  if $guardedmode; then
+    echo -n "${invers}git $gitcommand${reset} -> execute here (y/n)? "
+    read -n 1 -r </dev/tty; echo
+    if [[ $REPLY =~ ^[Yy]$ ]]; then atomicExecution; fi
+  else
+     atomicExecution
+  fi
+}
+
+# atomic git command execution with log
+function atomicExecution () {
+  echo "${bldred}->${reset} executing ${invers}git $gitcommand${reset}" && git $gitcommand
+}
+
+# check if the passed command is known as a core git command
+function checkGitCommand () {
+  if git help -a | cat | grep -o -q "\b${corecommand}\b"; then
+    echo "Core command \"$corecommand\" accepted."
+  else
+    if git config --get-regexp alias | grep -o -q "\.${corecommand} "; then
+      echo "Alias ${corecommand} accepted."
+    else
+      usage && echo "error: unknown GIT command: $corecommand" && exit 1
+    fi
+  fi
+}
+
+# check if workspace name is registered
+function checkWSName () {
+  found=0
+  while read workspace; do
+    rwsname=$(echo $workspace | cut -f1 -d' ' | cut -f2 -d'.')
+    if [[ $rwsname == $wsname ]]; then return; fi
+  done <<< "$(echo "$(listall)")"
+  # when here the ws name was not found
+  usage && echo "error: unknown workspace name: $wsname" && exit 1 
+}
+
+# detects the wsname of the current directory
+function wsnameToCurrent () {
+  currdir=$PWD
+  while read workspace; do
+    if [ -z "$workspace" ]; then continue; fi
+    rwsdir=$(echo $workspace | grep -o ' /.*')
+    rwsname=$(echo $workspace | cut -f1 -d' ' | cut -f2 -d'.')
```

Comment:
What about:
```
rwsdir=${workspace#* }
rwsname=${${workspace#*.}%% *}
```

And remember to use `local` with variable within function, to avoid polluting other scope.